internal/pkg/aws/s3/s3.go

```diff
@@ -52,3 +60,46 @@ func (s *Service) PutArtifact(bucket, fileName string, data io.Reader) (string,
 
 	return resp.Location, nil
 }
+
+// EmptyBucket deletes all objects within the bucket.
+func (s *Service) EmptyBucket(bucket string) error {
```

Comment:
Looks tricky, thanks for tackling this.

Have you looked at using the [BatchDelete](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/s3manager/#BatchDelete) of the s3Manager? 

Here's a [blog post](https://aws.amazon.com/blogs/developer/aws-sdk-for-go-batch-operations-with-amazon-s3/) on how it might help simplify things.

Reply:
Tried that method before. Ugh it is helpful when using non-versioning bucket. However, when it comes to "versioning bucket", things can be very complicated and "s3manager" they don't support versioning bucket very well, and `BatchDelete` cannot delete all the other versioning for the objects nor the "delete marker" for that.

Had a discussion with Efe yesterday on those resolved comments above on that.

Reply:
Ah, sorry then!

Reply:
lol no problem! 

@iamhopaul123  this looks so much neater and also correct :P thanks for looking into it